tests/providers/ftp/operators/test_ftp.py

```diff
@@ -184,3 +180,108 @@ def test_unequal_local_remote_file_paths(self):
                 local_filepath=["/tmp/test1", "/tmp/test2"],
                 remote_filepath="/tmp/test1",
             )
+
+
+class TestFTPSFileTransmitOperator:
+    def setup_method(self):
+        self.test_local_dir = "/tmp"
+        self.test_local_dir_int = "/tmp/interdir"
+        self.test_remote_dir = "/ftpshome"
+        self.test_remote_dir_int = "/ftpshome/interdir"
+        self.test_local_filename = "test_local_file"
+        self.test_remote_filename = "test_remote_file"
+        self.test_local_filepath = f"{self.test_local_dir}/{self.test_local_filename}"
+        self.test_remote_filepath = f"{self.test_remote_dir}/{self.test_remote_filename}"
+        self.test_local_filepath_int_dir = f"{self.test_local_dir_int}/{self.test_local_filename}"
+        self.test_remote_filepath_int_dir = f"{self.test_remote_dir_int}/{self.test_remote_filename}"
+
+    def teardown_method(self):
+        if os.path.exists(self.test_local_dir_int):
+            os.rmdir(self.test_local_dir_int)
```

Comment:
Are files/directories created during test? If so better to use `pytest`'s fixtures for that, rather than manually created them.
Depend of developer OS and configurations it might raise an errors as well as add some side effects for other tests.

Reply:
@Taragolis The testing code itself does not create any files or directories. Rather, for one of the tests (the `test_file_transfer_with_intermediate_dir_get`), the code that is being tested (from line 121 in `FTPFileTransmitOperator `in ftp.py) will create a new directory, so we need a teardown method to remove that created directory after that test. The FTPFileTransmitOperator is set up so that even if that directory already exists, it won't crash. Do you think it would be better to just mock out the creation of the directory from FTPFileTransmitOperator, so that during the tests no new directories are created?

Reply:
We almost complete migrate all tests from `unittests` to `pytest`.
Right now most of migration only about get rid of `unittests.TestCase` and their methods and we still use xunit-style for setup/teardown tests.

In general better to use all functional provided by `pytest` especially fixtures: provided by pytest, plugins, user-defined.
Out of the box pytest support two fixture which provide ability to work with temporary directories:
1. `tmp_path` for create unique temporary path (it could be also created during the tests)
2. `tmp_path_factory` for create temporary directories

see: https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
@Taragolis Thank you for letting me know, that is a good point. I have pushed a change where `tmp_path `is used and any directories that are created are created within this `tmp_path`. You can see the change in `test_file_transfer_with_intermediate_dir_get`. Let me know if there is anything else needed to be added.